xmppserver/src/main/java/org/jivesoftware/openfire/websocket/WebSocketClientConnectionHandler.java

```diff
@@ -281,7 +286,7 @@ private final class xmppSessionIdleTask extends TimerTask {
         @Override
         public void run()
         {
-            if (!isWebSocketOpen()) {
+            if (!isWebSocketOpen() || getMaxIdleTime().isNegative() || getMaxIdleTime().isZero()) {
```

Comment:
Is this correct logic? seems like these should all be !checks

Reply:
No,  I believe this is correct. This tries to _not_ run the check if the socket is closed, or if the max-idle time is non-positive (and thus disabled).